### PR TITLE
Scrape discussed pages and deprecate SCRAPE_WANTED

### PR DESCRIPTION
**Pull request by leogrim**
@eishay this ensures that discussed pages are scheduled for scraping

@martinraison @ymatsuda @yingjieMiao I noticed that the State[NormalizedURI] SCRAPE_WANTED could be removed with some care: we mostly rely on the existence of a ScrapeInfo to decide whether a page should be scraped so I saw an opportunity to further decouple the scraper logic from the NormalizedURIRepo. Thus I have deprecated SCRAPE_WANTED and effectively removed it from our code. This removes unnecessary writes to the NormalizedURI table but also reduces complexity: in order to scrape a uri, simply create a ScrapeInfo for it. 

I think the remaining states SCRAPED, UNSCRAPABLE and SCRAPE_FAILED could rather easily be migrated to the ScrapeInfo model itself (but I must wake up early tomorrow morning ;)

Thoughts?

### CODE REVIEW

File: kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala [hunk 1]
```diff
@@ -296,8 +296,7 @@ class FakeShoeboxServiceClientImpl(val airbrakeNotifier: AirbrakeNotifier) exten
   def getNormalizedUriByUrlOrPrenormalize(url: String): Future[Either[NormalizedURI, String]] = ???
 
 
-  def internNormalizedURI(urls: JsObject): Future[NormalizedURI] = {
-    val url = (urls \ "url").as[String]
+  def internNormalizedURI(url: String, scrapeWanted: Boolean): Future[NormalizedURI] = {
```
Comment (eishay) [inline kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala:299]: is scrapeWanted used in this method?

**Comment by eishay** [comment]
one less state is better :-) its already very hard to understand. hope it would make it easier to split scrape info to its own db

File: kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala (hunk 1 above)
Comment (leogrim) [inline kifi-backend/modules/common/test/com/keepit/shoebox/FakeShoeboxServiceClientImpl.scala:299]: No it's just the name of the boolean parameter, whether a ScrapeInfo should be created or not.

**Comment by leogrim** [comment]
@eishay I think it shouldn't be very hard at this point, at least we could finish up removing these states from NormalizedURI and adding them to ScrapeInfo. That would further alleviate writes on NormalizedURI and later on it would be easy to just move ScrapeInfo out. Whenever we should budget some time (a couple of days to move all logic to ScrapeInfo,a few more to get the table out?).

**Comment by ymatsuda** [comment]
Don't we need SCRAPE_WANTED state to split scrape_info to another db? Manipulating two db and keeping consistency is not trivial. Having SCRAPE_WANTED state is useful to implement eventual consistency.

**Comment by leogrim** [comment]
@ymatsuda I'm not sure I understand which consistency is at stake. Do you mean consistency between these two tables once they are in different databases? If the NormalizedURI table doesn't know anything about ScrapeInfo (no more scraping state), there is no more consistency challenge between these two. It seems to me that the real consistency to enforce is between Bookmark and ScrapeInfo, in order to make sure that every keep is scraped. This is what OrphanCleaner checks though. 

It seems to me that more decoupling between the part that deals with pages' uris and the part that deals with their content is a good thing. Am I missing something obvious?

**Comment by ymatsuda** [comment]
I got your point. You are right. We can synchronize scrape_info table from bookmark table. No need to go through normalized_uri.

**Comment by yingjieMiao** [comment]
lgtm

**Comment by raymondng** [comment]
:+1: Like the idea about skipping n_uri table. In earlier discussion wrt prioritization we considered skipping n_uri table (because it actually lacks information available in bookmark table) so I think this is the way to go.

Of course the fewer states the better!

**Pull request merged**